### PR TITLE
384 change sort order for resources

### PR DESCRIPTION
**Pull request by ViuCoding**

### CODE REVIEW

File: front/src/hooks/useSortByDate.tsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
+import { useEffect, useState } from 'react'
+
+type TTopic = {
```
Comment (kevinmamaqi) [inline front/src/hooks/useSortByDate.tsx:3]: type SortOrder = 'asc' | 'desc'

File: front/src/hooks/useSortByDate.tsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    downvote: number
+    total: number
+  }
+  resourceType: string
+  topics: TTopic[]
+}
+
+export const useSortByDate = (
+  items: TResource[] | undefined,
+  sortOrder: 'ascending' | 'descending' = 'descending'
```
Comment (kevinmamaqi) [inline front/src/hooks/useSortByDate.tsx:36]: sortOrder: SortOrder = 'desc'

File: front/src/hooks/useSortByDate.tsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  topic: {
+    id: string
+    name: string
+    slug: string
+    categoryId: string
+    createdAt: string
+    updatedAt: string
+  }
+}
+export type TResource = {
```
Comment (kevinmamaqi) [inline front/src/hooks/useSortByDate.tsx:13]: Eliminar type TResource es demasiado especifico

File: front/src/hooks/useSortByDate.tsx [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+  voteCount: {
+    upvote: number
+    downvote: number
+    total: number
+  }
+  resourceType: string
+  topics: TTopic[]
+}
+
+export const useSortByDate = (
```
Comment (kevinmamaqi) [inline front/src/hooks/useSortByDate.tsx:34]: export const useSortByDate = <T extends object>(
  items: T[] | undefined,
  sortBy: keyof T,

**Review by kevinmamaqi — CHANGES_REQUESTED** [review]
Me parece poco reutilizable

File: front/src/hooks/useSortByDate.tsx [hunk 5]
```diff
@@ -0,0 +1,30 @@
+import { useEffect, useState } from 'react'
+
+type SortOrder = 'asc' | 'desc'
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
```
Comment (kevinmamaqi) [inline front/src/hooks/useSortByDate.tsx:5]:
```suggestion

```

File: front/src/hooks/useSortByDate.tsx [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+export const useSortByDate = <T extends Record<string, any>>(
```
Comment (kevinmamaqi) [inline front/src/hooks/useSortByDate.tsx:6]:
```suggestion
export const useSortByDate = <T extends Record<string, unknown>>(
```

**Review by kevinmamaqi — CHANGES_REQUESTED** [review]

File: front/src/hooks/useSortByDate.tsx (hunk 6 above)
Comment (ViuCoding) [reply front/src/hooks/useSortByDate.tsx]: Hostras, sorry, arreglado, he abusado del "quick fix" =|

**Review by kevinmamaqi — APPROVED** [review]

**Pull request merged**